data/core/start.lua

```diff
@@ -19,3 +19,18 @@ package.path = DATADIR .. '/?.lua;' .. package.path
 package.path = DATADIR .. '/?/init.lua;' .. package.path
 package.path = USERDIR .. '/?.lua;' .. package.path
 package.path = USERDIR .. '/?/init.lua;' .. package.path
+
+local dynamic_suffix = MACOS and 'lib' or (WINDOWS and 'dll' or 'so')
+package.cpath = DATADIR .. '/?.' .. dynamic_suffix .. ';' .. package.cpath
+package.cpath = USERDIR .. '/?.' .. dynamic_suffix .. ';' .. package.cpath
+package.searchers[3] = function(modname)
```

Comment:
`package.searchers[4]` should be removed too unless it's explicitly needed to ensure all native module requires go through native plugin machinery.